.changeset/nine-mugs-accept.md

```diff
@@ -0,0 +1,79 @@
+---
+'@iroha2/crypto-target-bundler': major
+'@iroha2/crypto-target-web': major
+'@iroha2/crypto-core': major
+'@iroha2/crypto-target-node': major
+---
+
+**feature**: re-write WASM and provide high-level wrappers around it.
+
+### What the braking change is
+
+Each target now provides high-level wrappers around raw `wasm-pack` artifacts. These wrappers provide a better-designed interface with features like global `.free()`-objects tracking and integration with `@iroha2/data-model`.
+
+Moreover, the WASM itself is re-written and now provides more flexibility, such as working with JSON and HEX representations out of the box.
+
+The connection between `@iroha2/crypto-core` and `@iroha2/crypto-target-*` you can see here:
+
+```ts
+import { IrohaCryptoInterface, cryptoTypes } from '@iroha2/crypto-core'
+import { crypto } from '@iroha2/crypto-target-node'
+
+// each target exports `crypto` which is the `IrohaCryptoInterface` type from
+// the core library
+const cryptoAsserted: IrohaCryptoInterface = crypto
+
+// the core library exports `cryptoTypes` namespace which contains all the types
+// used in crypto you might need
+const hash: cryptoTypes.Hash = crypto.Hash.hash('hex', '00ff')
+```
+
+`@iroha2/crypto-core` re-exports `@iroha2/crypto-util`, a new library which contains (for now) only utilities to work with `.free()` tracking:
+
+```ts
+import { freeScope, FREE_HEAP } from '@iroha2/crypto-util'
+import { crypto } from '@iroha2/crypto-target-web'
+
+const keyPair = freeScope((scope) => {
+  const pair = crypto.KeyGenConfiguration
+    // here the configuration object is created, and you can `.free()` it manually
+    // fortunately, it is automatically attached to the scope it is created within,
+    // and when the scope is over, everything attached to it will be freed
```

Comment:
```suggestion
    // Create a configuration object that you can later `.free()` manually.
    // It is automatically attached to the scope it is created within,
    // so when the scope is over, everything attached to it will be freed.
```